setup.py

```diff
@@ -3,69 +3,55 @@
 # @Author: oesteban
 # @Date:   2015-11-19 16:44:27
 # @Last Modified by:   oesteban
-# @Last Modified time: 2016-08-29 18:50:54
+# @Last Modified time: 2016-09-13 11:36:06
 """ fmriprep setup script """
-import os
-import sys
+from io import open
 
 
 def main():
     """ Install entry-point """
+    from os import path as op
     from glob import glob
     from inspect import getfile, currentframe
     from setuptools import setup, find_packages
 
-    this_path = os.path.dirname(os.path.abspath(getfile(currentframe())))
-    # Read vars from info file
-    module_file =os.path.join(this_path, 'fmriprep', 'info.py')
-    with open(module_file) as fvars:
-       exec(compile(fvars.read(), module_file, 'exec'))
+    this_path = op.dirname(op.abspath(getfile(currentframe())))
 
-    REQ_LINKS = []
-    with open('requirements.txt', 'r') as rfile:
-        REQUIREMENTS = [line.strip() for line in rfile.readlines()]
-
-    for i, req in enumerate(REQUIREMENTS):
-        if req.startswith('-e'):
-            REQUIREMENTS[i] = req.split('=')[1]
-            REQ_LINKS.append(req.split()[1])
-
-    if REQUIREMENTS is None:
-        REQUIREMENTS = []
+    # Python 3: use a locals dictionary
+    # http://stackoverflow.com/a/1463370/6820620
+    ldict = locals()
+    # Get version and release info, which is all stored in fmriprep/info.py
+    module_file = op.join(this_path, 'fmriprep', 'info.py')
+    with open(module_file) as infofile:
+        pythoncode = [line for line in infofile.readlines() if not line.strip().startswith('#')]
+        exec('\n'.join(pythoncode), globals(), ldict)
```

Comment:
I actually started doing that but moved to this exec. Particularly, I was doing that in ```niworkflows```. The result was a chicken-and-egg problem: to get setup.py to the point of understanding nipype as a dependency, niworkflows required itself to be installed. Looking around this was the most effective solution.

If you try to import info.py from the setup.py you are assuming that the code has python package structure (which can change for various reasons). This is a more agnostic way to load those variables.